webapp/views/wpt-app.js

```diff
@@ -235,8 +242,45 @@ class WPTApp extends PathInfo(WPTFlags(TestRunsUIBase)) {
     };
     this.handleCollapse = () => {
       this.isBSFCollapsed = !this.isBSFCollapsed;
+      // Record hide/open actions on the BSF graph. Currently, we only
+      // show it on the homepage.
+      if ('ga' in window) {
+        window.ga('send', {
+          hitType: 'event',
+          eventCategory: 'bsf',
+          eventAction: 'hide',
+          eventLabel: this.path,
+          eventValue: this.isBSFCollapsed ? 1 : 0
+        });
+      }
       this.setLocalStorageFlag(this.isBSFCollapsed, 'isBSFCollapsed');
     };
+    this.enterBSF = () => {
```

Comment:
What exactly are we trying to measure here? It seems like we are measuring just the time spent in the general BSF area, not the chart specifically, and as such I don't get the purpose of the `isInteracting` variable.

Reply:
Correct, I want to measure the time spent in the general area because it would count the scenario when users look at the BSF graph without any interaction or only switch between stable and experimental.

`isInteracting` is a workaround for [this issue](https://stackoverflow.com/questions/17244996/why-do-the-mouseenter-mouseleave-events-fire-when-entering-leaving-child-element). where users interact with Google Chart, `enterBSF ` is triggered. I use `isInteracting` to avoid resetting `bsfStartTime`

Reply:
Oh this is super interesting! I was originally going to say I couldn't reproduce this, but then found that if I mouse quickly over the actual lines on the graph (maybe hitting the legend popups?) I can cause multiple `enterBSF`s to trigger - though without an `exitBSF` in the middle! E.g.

![image](https://user-images.githubusercontent.com/153264/104481066-51735b00-5593-11eb-98cc-edc15892e12c.png)

So I guess this LGTM, but please document `isInteracting`, it's very unclear why it exists!